### ISSUE TITLE
CMake does not check for duplicates before adding uninstall target

### ISSUE THREAD

**Issue by ribtoks** [Exiv2/exiv2#1217]
**Describe the bug**
If you several subdirectories in a bigger project add custom target called "uninstall", the build of the root project will fail because exiv2 does not check if the target exists.

**Expected behavior**
Use proper CMake code
```
if (NOT TARGET uninstall)
# only now add the target
endif()
```

**Desktop (please complete the following information):**
reproducible on any OS

**Comment by clanmills** [comment]
Thank You @ribtoks.  I believe we've spoken before. Are you the skype/microsoft guy in Lithuania.  No? 

As with #1216, I don't understand the use case and request our cmake guru @piponazo to review.

As with #1216.  This, if you

**Comment by piponazo** [comment]
Hi. I accepted and merged already the change proposed in #1216. I think this issue is a bit similar to the other one, in the sense that it is produced due to the way @ribtoks is consuming the project. I never thought about that use-case for consuming Exiv2 before, but I think it makes sense to fix these integration issues if they do not require a lot of work.

#1216 was merged only into master. @ribtoks are you interested in having both changes only on master, or would you like to have them also for the next 0.27.3 release ?

**Comment by ribtoks** [comment]
Hi. My first PR should be a safe cherry-pick for 0.27.3. There's no second change currently. Thanks!

**Comment by clanmills** [comment]
There are two issues #1216 and #1217.  #1216 has been merged and cherry-picked into 0.27-maintenance.

I'm not sure I've understood this uninstall issue.  Is this the patch:
```
526 rmills@rmillsmbp:~/gnu/github/exiv2/0.27-maintenance $ git diff
diff --git a/cmake/mainSetup.cmake b/cmake/mainSetup.cmake
index 1f8d10c8..31cdc6a2 100644
--- a/cmake/mainSetup.cmake
+++ b/cmake/mainSetup.cmake
@@ -26,6 +26,7 @@ if (UNIX)
         set(CMAKE_INSTALL_RPATH "$ORIGIN/../${CMAKE_INSTALL_LIBDIR}")
     endif()
 endif()
-
-configure_file(cmake/exiv2_uninstall.cmake ${CMAKE_BINARY_DIR}/cmake_uninstall.cmake COPYONLY)
-add_custom_target(uninstall "${CMAKE_COMMAND}" -P "${CMAKE_BINARY_DIR}/cmake_uninstall.cmake")
+if ( NOT TARGET uninstall )
+       configure_file(cmake/exiv2_uninstall.cmake ${CMAKE_BINARY_DIR}/cmake_uninstall.cmake COPYONLY)
+       add_custom_target(uninstall "${CMAKE_COMMAND}" -P "${CMAKE_BINARY_DIR}/cmake_uninstall.cmake")
+endif()
527 rmills@rmillsmbp:~/gnu/github/exiv2/0.27-maintenance $ 
```

**Comment by ribtoks** [comment]
@clanmills Yes, it looks like a patch for duplicate `uninstall` target issue.

**Comment by clanmills** [comment]
Ah, right.  I understand.  You've pushed both #1216 and the fix for #1217 into branch 0.27-maintenance-1216 and that's #1218.  Good Work, @piponazo Thank You.

**Comment by clanmills** [comment]
Uninstall will be investigate for v1.00: #1501
